src/transaction/transaction.rs

```diff
@@ -1031,4 +1130,100 @@ enum TransactionStatus {
     Rolledback,
     /// The transaction has tried to rollback. Only `rollback` is allowed.
     StartedRollback,
+    /// The transaction has been dropped.
+    Dropped,
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        mock::{MockKvClient, MockPdClient},
+        Transaction, TransactionOptions,
+    };
+    use fail::FailScenario;
+    use futures::executor::ThreadPool;
+    use std::{
+        any::Any,
+        io,
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc,
+        },
+    };
+    use tikv_client_proto::{kvrpcpb, pdpb::Timestamp};
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 6)]
+    async fn test_optimistic_heartbeat() -> Result<(), io::Error> {
+        let scenario = FailScenario::setup();
+        fail::cfg("after-prewrite", "return(true)").unwrap();
```

Comment:
why not use `fail::cfg("after-prewrite", "sleep(10000)").unwrap();`?

Reply:
It seems that `sleep(10000)` will block the threads so that the test will fail.

Reply:
So why it sleeps here:
https://github.com/tikv/client-rust/blob/b635096bed1eba224817ac827a056f18fad17898/tests/failpoints_tests.rs#L14